### PR TITLE
Created chart for openvpn

### PR DESCRIPTION
**Pull request by jfelten**
I though this might be useful to others so I thought I'd create a PR for it.

### CODE REVIEW

File: incubator/openvpn/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Helm chart for OpenVPN
+This chart will install an openvpn server inside a kubernetes cluster.  New certificates are generated on install, and a script is provided to genreate client keys as needed.  The chart will automatically configure dns to use kubeDNS and route all network traffic to kuberentes pods and services through the vpn.  By connecting to this VPN a host is effectively inside a cluster's network.
```
Comment (alejandroEsc) [inline incubator/openvpn/README.md]: r/genreate/generate

File: incubator/openvpn/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: extensions/v1beta1
+description: A Helm chart to install an openvpn server inside a kubernetes cluster.  Certificate generation is also part of the deployment, and this chart will genreate cleint keys as needed.
```
Comment (alejandroEsc) [inline incubator/openvpn/Chart.yaml]: r/genreate/generate

Comment (alejandroEsc) [reply incubator/openvpn/Chart.yaml]: r/ceint/client

File: incubator/openvpn/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		helm install incubator/openvpn
+
+Wait for the external load balancer IP to become available.  Check service status via: kubectl get svckubectl get svc
+ 
+Please be aware that certificate generation is variable and may take some time (minutes).
+check pod status via:
+POD_NAME=`kubectl get pods -l type=openvpn | awk END'{ print $1 }'` \
+&& kubectl log $POD_NAME --follow
+
+When ready generate a client key as follows:
```
Comment (alejandroEsc) [inline incubator/openvpn/README.md]: When ready, generate....

File: incubator/openvpn/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+		POD_NAME=`kubectl get pods -l type=openvpn | awk END'{ print $1 }'` \
+		&& SERVICE_NAME=`kubectl get svc -l type=openvpn | awk END'{ print $1 }'` \
+		&& SERVICE_IP=`kubectl get svc --namespace {{ .Release.Namespace }} $SERVICE_NAME -o jsonpath='{.status.loadBalancer.ingress[0].ip}'` \
+		&& KEY_NAME=kubeVPN \
+		&& kubectl exec -it $POD_NAME /etc/openvpn/setup/newClientCert.sh $KEY_NAME $SERVICE_IP \
+		&& kubectl exec -it $POD_NAME cat /usr/share/easy-rsa/pki/$KEY_NAME.ovpn > $KEY_NAME.ovpn
+
+Be sure to change KEY_NAME if genreating additional keys.  Import the .ovpn file into your favorite openvpn tool like tunnelblick and verify connectivity.
```
Comment (alejandroEsc) [inline incubator/openvpn/README.md]: r/genreating/generating

**Review by alejandroEsc — CHANGES_REQUESTED** [review]

**Comment by jfelten** [comment]
cc @spiffxp

**Comment by alejandroEsc** [comment]
@jfelten please squash your commits.

File: incubator/openvpn/templates/NOTES.txt [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+check service status via: kubectl get svckubectl get svc
+
+Once the external IP is available and all the server certificates are generated create client key .ovpn files by pasting the following into a shell:
+
+POD_NAME=`kubectl get pods -l type=openvpn | awk END'{ print $1 }'` \
+&& SERVICE_NAME=`kubectl get svc -l type=openvpn | awk END'{ print $1 }'` \
+&& SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} $SERVICE_NAME -o jsonpath='{.status.loadBalancer.ingress[0].ip}') \
+&& KEY_NAME=kubeVPN \
+&& kubectl exec -it $POD_NAME /etc/openvpn/setup/newClientCert.sh $KEY_NAME $SERVICE_IP \
+&& kubectl exec -it $POD_NAME cat /usr/share/easy-rsa/pki/$KEY_NAME.ovpn > $KEY_NAME.ovpn
```
Comment (viglesiasce) [inline incubator/openvpn/templates/NOTES.txt]: Not all of the kubectl commands in here are passing the namespace so this doesnt work in custom namespaces ATM.

**Review by viglesiasce — CHANGES_REQUESTED** [review]
One issue in the NOTES but this worked super well for me. Tested with Tunnelblick and was able to reach my Pods without issue.

Thank you so much for this awesome chart!

**Comment by viglesiasce** [comment]
@k8s-bot ok to test

**Review by viglesiasce — CHANGES_REQUESTED** [review]
Please move the chart to the stable folder

File: incubator/openvpn/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+The primary purpose of this chart was to make it easy to access kubernetes services during development.  It could also be used for any service that only needs to be accessed through a vpn or as a standard vpn.
+
+##Usage
+
+```bash
+helm repo add incubator http://storage.googleapis.com/kubernetes-charts-incubator
+helm install incubator/openvpn
+```
+
+Wait for the external load balancer IP to become available.  Check service status via: kubectl get svckubectl get svc
```
Comment (viglesiasce) [inline incubator/openvpn/README.md]: Typo here.

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: incubator/openvpn/templates/NOTES.txt (hunk 5 above)
Comment (jfelten) [reply incubator/openvpn/templates/NOTES.txt]: removed namespace - wasn't needed anyways

File: incubator/openvpn/README.md (hunk 6 above)
Comment (jfelten) [reply incubator/openvpn/README.md]: changed to stable

**Comment by viglesiasce** [comment]
Hey @jfelten, sorry for the confusion but I meant that all the commands needed the namespace added to them rather than removed.

**Comment by jfelten** [comment]
changed as requested

**Review by jfelten — COMMENTED** [review]
changed as requested

**Comment by viglesiasce** [comment]
@jfelten the exec commands also need the namespace flag

**Comment by jfelten** [comment]
I'm not sure the namespace restriction is necessary, but it doesn't hurt.  I added back to the appropriate kubectl commands.

**Comment by viglesiasce** [comment]
It is necessary when the namespace you deploy to is not the namespace in your current context.

**Comment by jfelten** [comment]
gotcha - hadn't considered that use case.  Added to namespace exec commands also.

**Review by viglesiasce — APPROVED** [review]

**Comment by viglesiasce** [comment]
Thanks @jfelten!

File: stable/openvpn/templates/config-openvpn.yaml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      
+      proto {{.Values.openvpn.OVPN_PROTO}}
+      port  {{ .Values.service.internalPort }}
+      dev tun0
+      status /tmp/openvpn-status.log
+      
+      user nobody
+      group nogroup
+      
+      push "route NETWORK 255.255.240.0"
```
Comment (so0k) [inline stable/openvpn/templates/config-openvpn.yaml:93]: what is the purpose of this route and netmask, this causes validation errors on linux vpn clients as the cidr is invalid @jfelten

**Pull request merged**